### PR TITLE
Code refactored to use linters, sec, vet and other go tools

### PR DESCRIPTION
**Pull request by enverbisevac**
Added CI action for checking and testing the code

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Set up Go
+      uses: actions/setup-go@v2
+      with:
+        go-version: 1.15
```
Comment (davejohnston) [inline .github/workflows/ci.yml:19]: 1.16 is available - I know it contains some security fixes over 1.15 so might be worth rev'ing to that.

**Review by davejohnston — COMMENTED** [review]
LGTM, minor comment about the golang version.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (enverbisevac) [reply .github/workflows/ci.yml:19]: Make sense

Comment (enverbisevac) [reply .github/workflows/ci.yml]: Done, and fixed unit tests

**Pull request merged**